store/postgres/src/subgraph_store.rs

```diff
@@ -923,6 +935,44 @@ impl SubgraphStoreInner {
         store.get_proof_of_indexing(site, indexer, block).await
     }
 
+    pub(crate) async fn get_public_proof_of_indexing(
+        &self,
+        id: &DeploymentHash,
+        block_number: BlockNumber,
+        block_store: Arc<impl BlockStore>,
+    ) -> Result<Option<(PartialBlockPtr, [u8; 32])>, StoreError> {
+        let (store, site) = self.store(&id).unwrap();
+
+        let chain_store = match block_store.chain_store(&site.network) {
+            Some(chain_store) => chain_store,
+            None => return Ok(None),
+        };
+        let mut hashes = chain_store.block_hashes_by_block_number(block_number)?;
+
+        // If we don't have this block or we have multiple versions of this block
+        // and using any of them could introduce non-deterministic because we don't
+        // know which one is the right one -> return no block hash
+        let block = PartialBlockPtr {
+            number: block_number,
+            hash: if hashes.is_empty() || hashes.len() > 1 {
+                None
+            } else {
+                Some(BlockHash::from(hashes.pop().unwrap()))
+            },
+        };
+
+        let block_for_poi_query = BlockPtr::new(
+            block.hash.clone().unwrap_or(BlockHash::zero()),
```

Comment:
Yeah, I agree.

Reply:
We can't expect clients to provide the hash, because the hash for the block number provided by the client may be different from the one that was processed by the node. So we'll have to deal with `Ok(None)`.